src/components/KTableView/KTableView.vue

```diff
@@ -603,6 +605,18 @@ const slots = useSlots()
 const tableId = useUniqueId()
 const { getSizeFromString } = useUtilities()
 
+const getRowKey = (row: Record<string, any>): string => {
+  if (typeof props.rowKey === 'function' && typeof props.rowKey(row) === 'string') {
+    return props.rowKey(row)
+  }
+
+  if (typeof props.rowKey === 'string' && !!(props.rowKey in row) && typeof row[props.rowKey] === 'string') {
+    return row[props.rowKey]
+  }
+
+  return ''
```

Comment:
question/issue(?): there are some places where we compare the result of this `getRowKey` function.

e.g.

```ts
const row = newData.find((dataRow) => getRowKey(selectedRow) === getRowKey(dataRow))
```

If there is a scenario where both calls evaluate to an empty string, then usage like shown above will always evaluate to `true`.

- Is that going to cause other issues?
- Should we be checking that `getRowKey` doesn't evaluate to an empty string in some/all instances?

Reply:
That entire block is [wrapped in a condition check](https://github.com/Kong/kongponents/blob/493320adb4c52f865e3e4da49d92226ae296fbf5/src/components/KTableView/KTableView.vue#L1261) that accounts for checking whether `getRowKey(row)` returns a truthy value

Reply:
Unless I'm missing something, there are still other places that aren't wrapped?

e.g. `getRowState`

Reply:
In other places we don't care if `getRowKey` returns an empty string since in **all instances** we are comparing it to `dataSelectState[n].rowKey` which is never going to be an empty string